common/v2/services/UnstoppableService/index.ts

```diff
@@ -0,0 +1,28 @@
+import Resolution, { ResolutionError } from '@unstoppabledomains/resolution';
+
+class UnstoppableResolution {
+  private resolution: Resolution;
+
+  constructor() {
+    this.resolution = new Resolution();
+  }
+
+  public async getResolvedAddress(domain: string, currencyTicker: string): Promise<string> {
+    return await this.resolution.addressOrThrow(domain, currencyTicker);
+  }
+
+  public isValidDomain(domain: string): boolean {
+    return this.resolution.isSupportedDomain(domain);
+  }
+
+  public parseResolutionError(err: ResolutionError, callback: any, params?: any) {
+    const setResolutionError = callback;
+    const { ticker, name } = params;
+    setResolutionError({
+      code: err.code,
+      params: { $domain: name, $recordName: ticker, $currencyTicker: ticker }
+    });
+  }
+}
```

Comment:
There seems to be no reason to execute the callback here.
Prefer a pure function that returns the formatted object, and execute the side-effect directly within the component.